app/routes/signups.js

```
@@ -0,0 +1,64 @@
+'use strict';
+
+const express = require('express');
+const router = express.Router(); // eslint-disable-line new-cap
+const CampaignNotFoundError = require('../exceptions/CampaignNotFoundError');
+const logger = app.locals.logger;
+
+router.post('/', (req, res) => {
+  if (!req.body.id) {
+    res.status(422).send({ error: 'id is required' });
+  }
+  const signupID = req.body.id;
+
+  if (!req.body.source) {
+    res.status(422).send({ error: 'source is required' });
+  }
+  const source = req.body.source;
+
+  logger.debug(`POST /v1/signups { id:${signupID}, source:${source} }`);
+
+  if (source === process.env.DS_API_POST_SOURCE) {
+    return res.send(`Already sent confirmation for signup ${signupID} with source ${source}.`);
+  }
+
+  let currentSignup;
+
+  return app.locals.db.signups
+    .lookupByID(signupID)
+    .then((signup) => {
+      currentSignup = signup;
+      if (!app.locals.campaigns[signup.campaign]) {
+        throw new CampaignNotFoundError();
+      }
+
+      return app.locals.db.users.lookup('id', currentSignup.user);
+    })
+    .then((user) => {
+      if (!user) {
+        return res.status(500).send({ error: 'Cannot find user for signup' });
```

Wouldn't it be passed to next `.then((user)`? Maybe throw an error instead?